src/gui/forcefieldTab.cpp

```diff
@@ -479,7 +543,40 @@ void ForcefieldTab::on_MasterTermAddBondButton_clicked(bool checked)
     dissolveWindow_->setModified();
 }
 
-void ForcefieldTab::on_MasterTermRemoveBondButton_clicked(bool checked) { Messenger::error("NOT IMPLEMENTED YET.\n"); }
+void ForcefieldTab::on_MasterTermRemoveBondButton_clicked(bool checked)
+{
+    QMessageBox queryBox;
+    queryBox.setWindowTitle(QString("Remove Master Bond"));
+    queryBox.setText(QString("Are you sure? This operation cannot be undone!\n") +
+                     QString("Any uses of this term in species will be converted to a local definition."));
+    queryBox.setInformativeText("Proceed?");
+    queryBox.setStandardButtons(QMessageBox::Yes | QMessageBox::No);
+    queryBox.setDefaultButton(QMessageBox::No);
+
+    if (queryBox.exec() != QMessageBox::Yes)
+        return;
+
+    if (dissolve_.coreData().nMasterBonds() == 1)
+        ui_.MasterTermRemoveBondButton->setEnabled(false);
+
+    auto index = ui_.MasterBondsTable->currentIndex();
+    if (!index.isValid())
+        return;
```

Comment:
Need to do the check for a valid index first so we don't needlessly create & present user choice. Also, not sure what the `if (dissolve_.coreData().nMasterBonds() == 1)` was achieving, so removed it! This change can propagate to the other term types.
```suggestion
    auto index = ui_.MasterBondsTable->currentIndex();
    if (!index.isValid())
        return;

    QMessageBox queryBox;
    queryBox.setWindowTitle(QString("Remove Master Bond"));
    queryBox.setText(QString("Are you sure? This operation cannot be undone!\n") +
                     QString("Any uses of this term in species will be converted to a local definition."));
    queryBox.setInformativeText("Proceed?");
    queryBox.setStandardButtons(QMessageBox::Yes | QMessageBox::No);
    queryBox.setDefaultButton(QMessageBox::No);

    if (queryBox.exec() != QMessageBox::Yes)
        return;
```

Reply:
`if (dissolve_.coreData().nMasterBonds() == 1)` was meant to disable the remove button if we are removing the last bond/angle/... etc.
Did this not solve an issue we had previously discussed where the remove button remained enabled after removing all items?

Reply:
Oh, I see - sorry.  Not really with it in the logic sense today!

Let's go with this solution for now, but hopefully will be able to revert it in the follow-up PR.